solana/indexer/damm_v2/indexer_test.go

```diff
@@ -0,0 +1,278 @@
+	require.NoError(t, err)
+
+	update := pb.SubscribeUpdate{
+		Filters: []string{address.String()},
+		UpdateOneof: &pb.SubscribeUpdate_Account{
+			Account: &pb.SubscribeUpdateAccount{
+				Account: &pb.SubscribeUpdateAccountInfo{
+					Pubkey: address.Bytes(),
+					Data:   positionData,
+				},
+			},
+		},
+	}
+
+	err = indexer.HandleUpdate(t.Context(), &update)
+	require.NoError(t, err)
+
+	rows, err := pool.Query(t.Context(), `
+		SELECT EXISTS (
+			SELECT 1
+			FROM sol_meteora_damm_v2_positions
+			JOIN sol_meteora_damm_v2_position_metrics ON sol_meteora_damm_v2_position_metrics.position = sol_meteora_damm_v2_positions.account
+			LIMIT 1
+		)
+	`)
+	require.NoError(t, err)
+	defer rows.Close()
+}
+
+type grpcClientMock struct {
+	mock.Mock
+
+	onUpdate common.DataCallback
+}
+
+func (m *grpcClientMock) Subscribe(ctx context.Context, req *pb.SubscribeRequest, onUpdate common.DataCallback, onError common.ErrorCallback) error {
+	args := m.Called(ctx, req, onUpdate, onError)
+	m.onUpdate = onUpdate
+	return args.Error(0)
+}
+
+func (m *grpcClientMock) Close() {
+	m.Called()
+}
+
+func TestSubscription(t *testing.T) {
+	pool := database.CreateTestDatabase(t, "test_solana_indexer_damm_v2")
+
+	// Fake an update for a Position and a Pool with missing data (should fail)
+	positionAddress := solana.MustPublicKeyFromBase58("5bYLydDXt1K5zroychcbrVbhGRUpheXdq5w41uccazPB")
+	positionUpdate := pb.SubscribeUpdate{
+		Filters: []string{positionAddress.String()},
+		UpdateOneof: &pb.SubscribeUpdate_Account{
+			Account: &pb.SubscribeUpdateAccount{
+				Account: &pb.SubscribeUpdateAccountInfo{
+					Pubkey: positionAddress.Bytes(),
+				},
+			},
+		},
+	}
+	dammPoolAddress := solana.MustPublicKeyFromBase58("D9iJqMbgQJLFt5PAAiTJTMNsMAMueukzoe1EK2r1g3WH")
+	poolUpdate := pb.SubscribeUpdate{
+		Filters: []string{NAME},
+		UpdateOneof: &pb.SubscribeUpdate_Account{
+			Account: &pb.SubscribeUpdateAccount{
+				Account: &pb.SubscribeUpdateAccountInfo{
+					Pubkey: dammPoolAddress.Bytes(),
+				},
+			},
+		},
+	}
+	dammPoolAddress2 := solana.MustPublicKeyFromBase58("8Z8rCYLuUcLfAJYPZxMgn6i9ifg9znQxrckXgZh6kYvN")
+
+	database.Seed(pool, database.FixtureMap{
+		"artist_coins": []map[string]any{
+			{
+				"mint":         "abc",
+				"ticker":       "",
+				"user_id":      0,
+				"decimals":     9,
+				"damm_v2_pool": dammPoolAddress.String(),
+			},
+		},
+	})
+
+	rpcClient := fake_rpc_client.FakeRpcClient{}
+	logger := zap.NewNop()
+
+	grpcMock := grpcClientMock{}
+	grpcMock.On("Subscribe", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(nil)
+	grpcMock.On("Close").Return()
+
+	indexer := New(common.GrpcConfig{}, &rpcClient, pool, logger)
+	indexer.grpcFactory = func(config common.GrpcConfig) common.GrpcClient {
+		return &grpcMock
+	}
+
+	ctx, cancel := context.WithTimeout(t.Context(), time.Second*5)
+	defer cancel()
+	go indexer.Start(ctx)
+
+	for {
+		if grpcMock.onUpdate != nil {
+			break
+		}
+		time.Sleep(time.Millisecond * 10)
```

Comment:
Can we make this sleep time any shorter? Or does it hurt things to just tight loop it? I'm wary of tests that add artificial timeouts.